Sources/DiscordModels/Types/Interaction.swift

```diff
@@ -10,52 +10,72 @@ public struct Interaction: Sendable, Codable {
-                case subCommand = 1
-                case subCommandGroup = 2
-                case string = 3
-                case integer = 4
-                case boolean = 5
-                case user = 6
-                case channel = 7
-                case role = 8
-                case mentionable = 9
-                case number = 10
-                case attachment = 11
+            /// https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-resolved-data-structure
+            public struct ResolvedData: Sendable, Codable {
+                public var users: [String: DiscordUser]?
+                public var members: [String: Guild.PartialMember]?
+                public var roles: [String: Role]?
+                public var channels: [String: PartialChannel]?
+                public var messages: [String: DiscordChannel.PartialMessage]?
+                public var attachments: [String: DiscordChannel.Message.Attachment]?
             }
             
+            /// https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-application-command-interaction-data-option-structure
+            public struct Option: Sendable, Codable {
+                
+                /// https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-option-type
+                public enum Kind: Int, Sendable, Codable, ToleratesIntDecodeMarker {
+                    case subCommand = 1
+                    case subCommandGroup = 2
+                    case string = 3
+                    case integer = 4
+                    case boolean = 5
+                    case user = 6
+                    case channel = 7
+                    case role = 8
+                    case mentionable = 9
+                    case number = 10
+                    case attachment = 11
+                }
+                
+                public var name: String
+                public var type: Kind
+                public var value: StringIntDoubleBool?
+                public var options: [Option]?
+                public var focused: Bool?
+            }
+            
+            public var id: String
             public var name: String
             public var type: Kind
-            public var value: StringIntDoubleBool?
+            public var resolved: ResolvedData?
             public var options: [Option]?
-            public var focused: Bool?
+            public var guild_id: String?
+            public var target_id: String?
+        }
+        
+        /// https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-message-component-data-structure
+        public struct MessageComponent: Sendable, Codable {
+            public var custom_id: String
+            public var component_type: ActionRow.Kind
+            public var values: [String]?
```

Comment:
You had set the type of this to be `[StringOrInt]?`.
Do you confirm the correct type is just `[String]?`?

Reply:
The type really seems to just be `[String]?` ... Since i want to release this changes, i'm going to resolve the conversation